### ISSUE TITLE
extern crate in local scope

### ISSUE THREAD

**Issue by aleksander** [rust-lang/rust#42776]
This code
```Rust
fn main() {
    extern crate libc;
    println!("Hello, world!");
}
```
compiled without any errors or warnings, but it's not possible to use any libc's items inside main. Should compiler warn about wrong place for `extern crate`? Or this is legal to use `extern crate` in local scope?

**Comment by kennytm** [comment]
#24219 → rust-lang/rfcs#959.

**Comment by Mark-Simulacrum** [comment]
Perhaps we should give a diagnostic message here though.. tagging A-diagnostics.

**Comment by steveklabnik** [comment]
Triage: now that most code will not use `extern crate`, and given that it's been over two years with no other people running into this, I'm going to give this a close. If someone wanted to work up a diagnostic, I don't think it would be denied, but I don't think this is ever realistically going to be fixed. Thank you!
